mbed-util/critical.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2015-2016, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __MBED_UTIL_CRITICAL_H__
+#define __MBED_UTIL_CRITICAL_H__
+
+#ifdef __cplusplus
+namespace mbed {
+namespace util {
+extern "C" {
```

Comment:
I don't believe this has any effect at all. The `extern "C"` will disable the namespacing, since it needs to follow C linkage rules:

```
>arm-none-eabi-readelf.exe -s critical.c.o

Symbol table '.symtab' contains 30 entries:
   Num:    Value  Size Type    Bind   Vis      Ndx Name
     0: 00000000     0 NOTYPE  LOCAL  DEFAULT  UND
     1: 00000000     0 FILE    LOCAL  DEFAULT  ABS critical.c
..........
    28: 00000001    40 FUNC    GLOBAL DEFAULT    4 critical_section_enter
    29: 00000001    40 FUNC    GLOBAL DEFAULT    6 critical_section_exit
```